lib/src/ssh_signing.rs

```diff
@@ -88,9 +88,10 @@ fn run_command(command: &mut Command, stdin: &[u8]) -> SshResult<Vec<u8>> {
 // If the given data is actually already a filepath to a key on disk then the
 // key input is returned directly.
 fn ensure_key_as_file(key: &str) -> SshResult<Either<PathBuf, tempfile::TempPath>> {
-    let is_inlined_ssh_key = key.starts_with("ssh-");
+    let key_path = crate::file_util::expand_home_path(key);
+
+    let is_inlined_ssh_key = !Path::is_absolute(&key_path);
     if !is_inlined_ssh_key {
```

Comment:
nit: `!key_path.is_absolute()` (also good to inline it to remove double negates.)